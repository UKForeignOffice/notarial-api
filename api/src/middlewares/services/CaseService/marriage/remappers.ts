import * as affirmationMappings from "./mappings/affirmation";
import * as cniPostalMappings from "./mappings/cniPostal";
import * as exchangeMappings from "./mappings/exchange";
import * as mscMappings from "./mappings/msc";
import * as cniMSCCombinedMappings from "./mappings/CNIAndMSC";
import { createRemapper } from "../utils/createRemapper";

export const remappers = {
  affirmation: createRemapper(affirmationMappings.remap, [], ["file"]),
  cni: createRemapper(affirmationMappings.remap, [], ["file"]),
  cniPostal: createRemapper(cniPostalMappings.remap, [], ["file"]),
  exchange: createRemapper(exchangeMappings.remap, [], ["file"]),
  mscPostal: createRemapper(mscMappings.remap, [], ["file"]),
  cniAndMscPostal: createRemapper(cniMSCCombinedMappings.remap, [], ["file"]),
};
