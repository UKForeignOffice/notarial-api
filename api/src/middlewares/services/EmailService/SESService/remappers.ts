import * as affirmationMappings from "./mappings/affirmation";
import { createRemapper } from "./utils/createRemapper";

export const remappers = {
  affirmation: createRemapper(affirmationMappings.remap, [], ["file"]),
};
