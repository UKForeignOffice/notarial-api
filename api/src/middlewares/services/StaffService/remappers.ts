import * as affirmationMappings from "./mappings/affirmation";
import * as exchangeMappings from "./mappings/exchange";
import { createRemapper } from "./utils/createRemapper";

export const remappers = {
  affirmation: createRemapper(affirmationMappings.remap, [], ["file"]),
  cni: createRemapper(affirmationMappings.remap, [], ["file"]),
  exchange: createRemapper(exchangeMappings.remap, [], ["file"]),
};
