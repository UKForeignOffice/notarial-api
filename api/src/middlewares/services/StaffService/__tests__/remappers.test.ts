import { remappers } from "../remappers";
import { fields } from "./fixtures/fields";

test("affirmation is successfully remapped", () => {
  const remapped = remappers.affirmation(fields);
  expect(remapped).toMatchSnapshot();
});

test("CNIAndMSC is successfully remapped", () => {
  const remapped = remappers.cniAndMscPostal(fields);
  expect(remapped).toMatchSnapshot();
});

test("cniPostal is successfully remapped", () => {
  const remapped = remappers.cniPostal(fields);
  expect(remapped).toMatchSnapshot();
});

test("exchange is successfully remapped", () => {
  const remapped = remappers.exchange(fields);
  expect(remapped).toMatchSnapshot();
});

test("msc is successfully remapped", () => {
  const remapped = remappers.mscPostal(fields);
  expect(remapped).toMatchSnapshot();
});
