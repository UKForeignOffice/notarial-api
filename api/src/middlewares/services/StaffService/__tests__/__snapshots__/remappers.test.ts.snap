// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CNIAndMSC is successfully remapped 1`] = `
{
  "applicantAddress": {
    "addressLine1": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "addressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "addressLine2": {
      "answer": null,
      "category": "applicantDetails",
      "key": "addressLine2",
      "title": "Address line 2 ",
      "type": "text",
    },
    "city": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "city",
      "title": "Town or city",
      "type": "text",
    },
    "country": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryAddress",
      "title": "Country",
      "type": "list",
    },
    "postcode": {
      "answer": null,
      "category": "applicantDetails",
      "key": "postcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "applicantDetails": {
    "dateOfBirth": {
      "answer": "1998-01-19",
      "category": "applicantDetails",
      "key": "dateOfBirth",
      "title": "Date of birth",
      "type": "date",
    },
    "emailAddress": {
      "answer": "luke@cautionyourblast.com",
      "category": "contact",
      "key": "emailAddress",
      "title": "Email address",
      "type": "text",
    },
    "firstName": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "firstName",
      "title": "First name",
      "type": "text",
    },
    "middleName": {
      "answer": null,
      "category": "applicantDetails",
      "key": "middleName",
      "title": "Middle name",
      "type": "text",
    },
    "phoneNumber": {
      "answer": "+44 123456787890",
      "category": "contact",
      "key": "phoneNumber",
      "title": "Phone number",
      "type": "text",
    },
    "sex": {
      "answer": "Male",
      "category": "applicantDetails",
      "key": "sex",
      "title": "Male or female",
      "type": "list",
    },
    "surname": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "surname",
      "title": "Surname",
      "type": "text",
    },
  },
  "applicantPassport": {
    "dateOfIssue": {
      "answer": "1998-01-19",
      "category": "applicantDetails",
      "key": "passportDateOfIssue",
      "title": "Date of issue of passport",
      "type": "date",
    },
    "number": {
      "answer": "1234567899",
      "category": "applicantDetails",
      "key": "passportNumber",
      "title": "Passport number",
      "type": "text",
    },
  },
  "countryOfBirth": {
    "answer": "United Kingdom",
    "category": "applicantDetails",
    "key": "countryOfBirth",
    "title": "Country of birth",
    "type": "list",
  },
  "declaration": {
    "answer": true,
    "key": "declaration",
    "title": "Declaration",
    "type": "boolean",
  },
  "fatherFullName": {
    "answer": "test",
    "key": "fatherFullName",
    "title": "Father's full name",
    "type": "text",
  },
  "feedbackConsent": {
    "feedbackConsent": {
      "answer": false,
      "category": "feedback",
      "key": "feedbackConsent",
      "title": "Can our partner contact you for feedback to help improve this service?",
      "type": "list",
    },
  },
  "information": {
    "country": {
      "answer": "Spain",
      "category": "marriage",
      "key": "country",
      "title": "Country",
      "type": "list",
    },
  },
  "livesInCountry": {
    "answer": true,
    "category": "applicantDetails",
    "key": "livesInCountry",
    "title": "Lives in the country",
    "type": "list",
  },
  "maritalStatus": {
    "status": {
      "answer": "Never married",
      "category": "maritalStatus",
      "key": "maritalStatus",
      "title": "Marital status",
      "type": "list",
    },
  },
  "marriageDetails": {
    "dateOfMarriage": {
      "answer": "2023-10",
      "category": "marriage",
      "key": "dateOfMarriage",
      "title": "Date of marriage",
      "type": "monthYear",
    },
    "placeOfMarriage": {
      "answer": "test",
      "category": "marriage",
      "key": "placeOfMarriage",
      "title": "Place of marriage",
      "type": "text",
    },
  },
  "motherFullName": {
    "answer": "test",
    "key": "motherFullName",
    "title": "Mother's full name",
    "type": "text",
  },
  "motherMaidenName": {
    "answer": "test",
    "key": "motherMaidenName",
    "title": "Maiden name",
    "type": "text",
  },
  "partnerAddress": {
    "partnerAddressCity": {
      "answer": "test",
      "category": "partner",
      "key": "partnerAddressCity",
      "title": "Town or city",
      "type": "text",
    },
    "partnerAddressCountry": {
      "answer": "United Kingdom",
      "category": "partner",
      "key": "partnerAddressCountry",
      "title": "Country",
      "type": "list",
    },
    "partnerAddressLine1": {
      "answer": "test",
      "category": "partner",
      "key": "partnerAddressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "partnerAddressLine2": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressLine2",
      "title": "Address line 2",
      "type": "text",
    },
    "partnerAddressPostcode": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressPostcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "partnerMaritalStatus": {
    "partnerMaritalStatus": {
      "answer": "Never married",
      "category": "partner",
      "key": "partnerMaritalStatus",
      "title": "Marital status",
      "type": "list",
    },
  },
  "partnerName": {
    "partnerFirstName": {
      "answer": "test",
      "category": "partner",
      "key": "partnerFirstName",
      "title": "Partner’s first name",
      "type": "text",
    },
    "partnerMiddleName": {
      "answer": null,
      "category": "partner",
      "key": "partnerMiddleName",
      "title": "Partner’s middle name",
      "type": "text",
    },
    "partnerSex": {
      "answer": "Male",
      "category": "partner",
      "key": "partnerSex",
      "title": "Partner’s sex",
      "type": "list",
    },
    "partnerSurname": {
      "answer": "testq",
      "category": "partner",
      "key": "partnerSurname",
      "title": "Partner’s surname",
      "type": "text",
    },
  },
  "partnerNationality": {
    "partnerNationality": {
      "answer": "British",
      "category": "partner",
      "key": "partnerNationality",
      "title": "Partner’s nationality",
      "type": "list",
    },
  },
  "placeOfBirth": {
    "answer": "test",
    "category": "applicantDetails",
    "key": "placeOfBirth",
    "title": "Place of birth ",
    "type": "text",
  },
  "service": {
    "answer": "cniAndMsc",
    "category": "documents",
    "key": "service",
    "title": "Service being applied for",
    "type": "list",
  },
}
`;

exports[`affirmation is successfully remapped 1`] = `
{
  "applicantAddress": {
    "addressLine1": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "addressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "addressLine2": {
      "answer": null,
      "category": "applicantDetails",
      "key": "addressLine2",
      "title": "Address line 2 ",
      "type": "text",
    },
    "city": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "city",
      "title": "Town or city",
      "type": "text",
    },
    "country": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryAddress",
      "title": "Country",
      "type": "list",
    },
    "postcode": {
      "answer": null,
      "category": "applicantDetails",
      "key": "postcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "applicantBirth": {
    "countryOfBirth": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryOfBirth",
      "title": "Country of birth",
      "type": "list",
    },
    "dateOfBirth": {
      "answer": "1998-01-19",
      "category": "applicantDetails",
      "key": "dateOfBirth",
      "title": "Date of birth",
      "type": "date",
    },
    "nameChangedByDeedPoll": {
      "answer": "false",
      "category": "nameChange",
      "key": "nameChangedByDeedPoll",
      "title": "Name changed by deed poll",
      "type": "list",
    },
    "placeOfBirth": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "placeOfBirth",
      "title": "Place of birth ",
      "type": "text",
    },
  },
  "applicantDetails": {
    "firstName": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "firstName",
      "title": "First name",
      "type": "text",
    },
    "middleName": {
      "answer": null,
      "category": "applicantDetails",
      "key": "middleName",
      "title": "Middle name",
      "type": "text",
    },
    "sex": {
      "answer": "Male",
      "category": "applicantDetails",
      "key": "sex",
      "title": "Male or female",
      "type": "list",
    },
    "surname": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "surname",
      "title": "Surname",
      "type": "text",
    },
  },
  "applicantPassport": {
    "dateOfIssue": {
      "answer": "1998-01-18",
      "category": "applicantDetails",
      "key": "passportDateOfIssue",
      "title": "Date of issue of passport",
      "type": "date",
    },
    "number": {
      "answer": "1234567890",
      "category": "applicantDetails",
      "key": "passportNumber",
      "title": "Passport number",
      "type": "text",
    },
  },
  "contactDetails": {
    "emailAddress": {
      "answer": "luke@cautionyourblast.com",
      "category": "contact",
      "key": "emailAddress",
      "title": "Email address",
      "type": "text",
    },
    "phoneNumber": {
      "answer": "+44 1234 567890",
      "category": "contact",
      "key": "phoneNumber",
      "title": "Phone number",
      "type": "text",
    },
  },
  "declaration": {
    "answer": true,
    "key": "declaration",
    "title": "Declaration",
    "type": "boolean",
  },
  "feedbackConsent": {
    "feedbackConsent": {
      "answer": false,
      "category": "feedback",
      "key": "feedbackConsent",
      "title": "Can our partner contact you for feedback to help improve this service?",
      "type": "list",
    },
  },
  "information": {
    "country": {
      "answer": "Belarus",
      "category": "marriage",
      "key": "country",
      "title": "Country",
      "type": "list",
    },
    "jurats": {
      "answer": "No",
      "category": "affirmationDoc",
      "key": "jurats",
      "title": "Accessibility needs",
      "type": "list",
    },
    "oathType": {
      "answer": "Non-religious",
      "category": "affirmationDoc",
      "key": "oathType",
      "title": "Type of oath",
      "type": "list",
    },
  },
  "maritalStatus": {
    "status": {
      "answer": "Never married",
      "category": "maritalStatus",
      "key": "maritalStatus",
      "title": "Marital status",
      "type": "list",
    },
  },
  "marriageDetails": {
    "dateOfMarriage": {
      "answer": "2023-10",
      "category": "marriage",
      "key": "dateOfMarriage",
      "title": "Date of marriage",
      "type": "monthYear",
    },
    "placeOfMarriage": {
      "answer": "test",
      "category": "marriage",
      "key": "placeOfMarriage",
      "title": "Place of marriage",
      "type": "text",
    },
  },
  "occupation": {
    "occupation": {
      "answer": "test",
      "category": "occupation",
      "key": "occupation",
      "title": "Occupation",
      "type": "text",
    },
  },
  "parentDetails": {
    "fatherFullName": {
      "answer": "test",
      "category": "parents",
      "key": "fatherFullName",
      "title": "Father's full name",
      "type": "text",
    },
    "motherFullName": {
      "answer": "test",
      "category": "parents",
      "key": "motherFullName",
      "title": "Mother's full name",
      "type": "text",
    },
    "motherMaidenName": {
      "answer": "test",
      "category": "parents",
      "key": "motherMaidenName",
      "title": "Maiden name",
      "type": "text",
    },
  },
  "partnerAddress": {
    "partnerAddressCity": {
      "answer": "test",
      "category": "partner",
      "key": "partnerAddressCity",
      "title": "Town or city",
      "type": "text",
    },
    "partnerAddressCountry": {
      "answer": "United Kingdom",
      "category": "partner",
      "key": "partnerAddressCountry",
      "title": "Country",
      "type": "list",
    },
    "partnerAddressLine1": {
      "answer": "test",
      "category": "partner",
      "key": "partnerAddressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "partnerAddressLine2": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressLine2",
      "title": "Address line 2",
      "type": "text",
    },
    "partnerAddressPostcode": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressPostcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "partnerName": {
    "partnerFirstName": {
      "answer": "test",
      "category": "partner",
      "key": "partnerFirstName",
      "title": "Partner’s first name",
      "type": "text",
    },
    "partnerMiddleName": {
      "answer": null,
      "category": "partner",
      "key": "partnerMiddleName",
      "title": "Partner’s middle name",
      "type": "text",
    },
    "partnerSex": {
      "answer": "Male",
      "category": "partner",
      "key": "partnerSex",
      "title": "Partner’s sex",
      "type": "list",
    },
    "partnerSurname": {
      "answer": "test",
      "category": "partner",
      "key": "partnerSurname",
      "title": "Partner’s surname",
      "type": "text",
    },
  },
  "partnerNationality": {
    "partnerNationality": {
      "answer": "British",
      "category": "partner",
      "key": "partnerNationality",
      "title": "Partner’s nationality",
      "type": "list",
    },
  },
}
`;

exports[`cniPostal is successfully remapped 1`] = `
{
  "applicantAddress": {
    "addressLine1": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "addressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "addressLine2": {
      "answer": null,
      "category": "applicantDetails",
      "key": "addressLine2",
      "title": "Address line 2 ",
      "type": "text",
    },
    "city": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "city",
      "title": "Town or city",
      "type": "text",
    },
    "country": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryAddress",
      "title": "Country",
      "type": "list",
    },
    "postcode": {
      "answer": null,
      "category": "applicantDetails",
      "key": "postcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "applicantDetails": {
    "emailAddress": {
      "answer": "luke@cautionyourblast.com",
      "category": "contact",
      "key": "emailAddress",
      "title": "Email address",
      "type": "text",
    },
    "firstName": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "firstName",
      "title": "First name",
      "type": "text",
    },
    "middleName": {
      "answer": null,
      "category": "applicantDetails",
      "key": "middleName",
      "title": "Middle name",
      "type": "text",
    },
    "phoneNumber": {
      "answer": "+44 1234 567890",
      "category": "contact",
      "key": "phoneNumber",
      "title": "Phone number",
      "type": "text",
    },
    "surname": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "surname",
      "title": "Surname",
      "type": "text",
    },
  },
  "declaration": {
    "answer": true,
    "key": "declaration",
    "title": "Declaration",
    "type": "boolean",
  },
  "feedbackConsent": {
    "feedbackConsent": {
      "answer": false,
      "category": "feedback",
      "key": "feedbackConsent",
      "title": "Can our partner contact you for feedback to help improve this service?",
      "type": "list",
    },
  },
  "information": {
    "country": {
      "answer": "Spain",
      "category": "marriage",
      "key": "country",
      "title": "Country",
      "type": "list",
    },
  },
  "livesInCountry": {
    "answer": true,
    "category": "applicantDetails",
    "key": "livesInCountry",
    "title": "Lives in the country",
    "type": "list",
  },
  "maritalStatus": {
    "status": {
      "answer": "Never married",
      "category": "maritalStatus",
      "key": "maritalStatus",
      "title": "Marital status",
      "type": "list",
    },
  },
  "marriageDetails": {
    "dateOfMarriage": {
      "answer": "2023-10",
      "category": "marriage",
      "key": "dateOfMarriage",
      "title": "Date of marriage",
      "type": "monthYear",
    },
    "placeOfMarriage": {
      "answer": "test",
      "category": "marriage",
      "key": "placeOfMarriage",
      "title": "Place of marriage",
      "type": "text",
    },
  },
  "partnerAddress": {
    "partnerAddressCity": {
      "answer": "test",
      "category": "partner",
      "key": "partnerAddressCity",
      "title": "Town or city",
      "type": "text",
    },
    "partnerAddressCountry": {
      "answer": "United Kingdom",
      "category": "partner",
      "key": "partnerAddressCountry",
      "title": "Country",
      "type": "list",
    },
    "partnerAddressLine1": {
      "answer": "test",
      "category": "partner",
      "key": "partnerAddressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "partnerAddressLine2": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressLine2",
      "title": "Address line 2",
      "type": "text",
    },
    "partnerAddressPostcode": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressPostcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "partnerMaritalStatus": {
    "partnerMaritalStatus": {
      "answer": "Never married",
      "category": "partner",
      "key": "partnerMaritalStatus",
      "title": "Marital status",
      "type": "list",
    },
  },
  "partnerName": {
    "partnerFirstName": {
      "answer": "test",
      "category": "partner",
      "key": "partnerFirstName",
      "title": "Partner’s first name",
      "type": "text",
    },
    "partnerMiddleName": {
      "answer": null,
      "category": "partner",
      "key": "partnerMiddleName",
      "title": "Partner’s middle name",
      "type": "text",
    },
    "partnerSex": {
      "answer": "Male",
      "category": "partner",
      "key": "partnerSex",
      "title": "Partner’s sex",
      "type": "list",
    },
    "partnerSurname": {
      "answer": "test",
      "category": "partner",
      "key": "partnerSurname",
      "title": "Partner’s surname",
      "type": "text",
    },
  },
  "partnerNationality": {
    "partnerNationality": {
      "answer": "British",
      "category": "partner",
      "key": "partnerNationality",
      "title": "Partner’s nationality",
      "type": "list",
    },
  },
  "service": {
    "answer": "cni",
    "category": "documents",
    "key": "service",
    "title": "Service being applied for",
    "type": "list",
  },
}
`;

exports[`exchange is successfully remapped 1`] = `
{
  "applicantAddress": {
    "addressLine1": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "addressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "addressLine2": {
      "answer": null,
      "category": "applicantDetails",
      "key": "addressLine2",
      "title": "Address line 2 ",
      "type": "text",
    },
    "city": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "city",
      "title": "Town or city",
      "type": "text",
    },
    "country": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryAddress",
      "title": "Country",
      "type": "list",
    },
    "postcode": {
      "answer": null,
      "category": "applicantDetails",
      "key": "postcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "applicantDetails": {
    "emailAddress": {
      "answer": "luke@cautionyourblast.com",
      "category": "contact",
      "key": "emailAddress",
      "title": "Email address",
      "type": "text",
    },
    "firstName": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "firstName",
      "title": "First name",
      "type": "text",
    },
    "middleName": {
      "answer": null,
      "category": "applicantDetails",
      "key": "middleName",
      "title": "Middle name",
      "type": "text",
    },
    "phoneNumber": {
      "answer": "+44 1234 567890",
      "category": "contact",
      "key": "phoneNumber",
      "title": "Phone number",
      "type": "text",
    },
    "surname": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "surname",
      "title": "Surname",
      "type": "text",
    },
  },
  "declaration": {
    "answer": true,
    "key": "declaration",
    "title": "Declaration",
    "type": "boolean",
  },
  "feedbackConsent": {
    "feedbackConsent": {
      "answer": false,
      "category": "feedback",
      "key": "feedbackConsent",
      "title": "Can our partner contact you for feedback to help improve this service?",
      "type": "list",
    },
  },
  "information": {
    "certifyPassport": {
      "answer": true,
      "category": "additionalServices",
      "key": "certifyPassport",
      "title": "Do you want to certify your passport as part of your CNI exchange application?",
      "type": "list",
    },
    "country": {
      "answer": "Sweden",
      "category": "marriage",
      "key": "country",
      "title": "Country",
      "type": "list",
    },
  },
  "maritalStatus": {
    "status": {
      "answer": "Never married",
      "category": "applicantDetails",
      "key": "maritalStatus",
      "title": "Marital status",
      "type": "list",
    },
  },
  "marriageDetails": {
    "dateOfMarriage": {
      "answer": "2203-10",
      "category": "marriage",
      "key": "dateOfMarriage",
      "title": "Date of marriage",
      "type": "monthYear",
    },
    "placeOfMarriage": {
      "answer": "test",
      "category": "marriage",
      "key": "placeOfMarriage",
      "title": "Place of marriage",
      "type": "text",
    },
  },
  "partnerAddress": {
    "partnerAddressCity": {
      "answer": "test",
      "category": "partner",
      "key": "partnerAddressCity",
      "title": "Town or city",
      "type": "text",
    },
    "partnerAddressCountry": {
      "answer": "United Kingdom",
      "category": "partner",
      "key": "partnerAddressCountry",
      "title": "Country",
      "type": "list",
    },
    "partnerAddressLine1": {
      "answer": "test",
      "category": "partner",
      "key": "partnerAddressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "partnerAddressLine2": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressLine2",
      "title": "Address line 2",
      "type": "text",
    },
    "partnerAddressPostcode": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressPostcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "partnerMaritalStatus": {
    "partnerMaritalStatus": {
      "answer": "Never married",
      "category": "partner",
      "key": "partnerMaritalStatus",
      "title": "Partner’s marital status",
      "type": "list",
    },
  },
  "partnerName": {
    "partnerFirstName": {
      "answer": "test",
      "category": "partner",
      "key": "partnerFirstName",
      "title": "Partner’s first name",
      "type": "text",
    },
    "partnerMiddleName": {
      "answer": null,
      "category": "partner",
      "key": "partnerMiddleName",
      "title": "Partner’s middle name",
      "type": "text",
    },
    "partnerSex": {
      "answer": "Male",
      "category": "partner",
      "key": "partnerSex",
      "title": "Partner’s sex",
      "type": "list",
    },
    "partnerSurname": {
      "answer": "test",
      "category": "partner",
      "key": "partnerSurname",
      "title": "Partner’s surname",
      "type": "text",
    },
  },
  "partnerNationality": {
    "partnerNationality": {
      "answer": "British",
      "category": "partner",
      "key": "partnerNationality",
      "title": "Partner’s nationality",
      "type": "list",
    },
  },
}
`;

exports[`msc is successfully remapped 1`] = `
{
  "applicantDetails": {
    "dateOfBirth": {
      "answer": "1998-01-19",
      "category": "applicantDetails",
      "key": "dateOfBirth",
      "title": "Date of birth",
      "type": "date",
    },
    "emailAddress": {
      "answer": "luke@cautionyourblast.com",
      "category": "contact",
      "key": "emailAddress",
      "title": "Email address",
      "type": "text",
    },
    "phoneNumber": {
      "answer": "+44 12345678900",
      "category": "contact",
      "key": "phoneNumber",
      "title": "Phone number",
      "type": "text",
    },
    "sex": {
      "answer": "Male",
      "category": "applicantDetails",
      "key": "sex",
      "title": "Male or female",
      "type": "list",
    },
  },
  "applicantPassport": {
    "dateOfIssue": {
      "answer": "1998-01-19",
      "category": "applicantDetails",
      "key": "passportDateOfIssue",
      "title": "Date of issue of passport",
      "type": "date",
    },
    "number": {
      "answer": "1234567890",
      "category": "applicantDetails",
      "key": "passportNumber",
      "title": "Passport number",
      "type": "text",
    },
  },
  "countryOfBirth": {
    "answer": "United Kingdom",
    "category": "applicantDetails",
    "key": "countryOfBirth",
    "title": "Country of birth",
    "type": "list",
  },
  "declaration": {
    "answer": true,
    "key": "declaration",
    "title": "Declaration",
    "type": "boolean",
  },
  "fatherFullName": {
    "answer": "test",
    "key": "fatherFullName",
    "title": "Father's full name",
    "type": "text",
  },
  "feedbackConsent": {
    "feedbackConsent": {
      "answer": false,
      "category": "feedback",
      "key": "feedbackConsent",
      "title": "Can our partner contact you for feedback to help improve this service?",
      "type": "list",
    },
  },
  "firstName": {
    "answer": "test",
    "category": "applicantDetails",
    "key": "firstName",
    "title": "First name",
    "type": "text",
  },
  "information": {
    "country": {
      "answer": "Spain",
      "category": "marriage",
      "key": "country",
      "title": "Country",
      "type": "list",
    },
  },
  "livesInCountry": {
    "answer": true,
    "category": "applicantDetails",
    "key": "livesInCountry",
    "title": "Lives in the country",
    "type": "list",
  },
  "maritalStatus": {
    "status": {
      "answer": "Never married",
      "category": "maritalStatus",
      "key": "maritalStatus",
      "title": "Marital status",
      "type": "list",
    },
  },
  "marriageDetails": {
    "dateOfMarriage": {
      "answer": "2023-10",
      "category": "marriage",
      "key": "dateOfMarriage",
      "title": "Date of marriage",
      "type": "monthYear",
    },
    "placeOfMarriage": {
      "answer": "test",
      "category": "marriage",
      "key": "placeOfMarriage",
      "title": "Place of marriage",
      "type": "text",
    },
  },
  "middleName": {
    "answer": null,
    "category": "applicantDetails",
    "key": "middleName",
    "title": "Middle name",
    "type": "text",
  },
  "motherFullName": {
    "answer": "test",
    "key": "motherFullName",
    "title": "Mother's full name",
    "type": "text",
  },
  "motherMaidenName": {
    "answer": "test",
    "key": "motherMaidenName",
    "title": "Maiden name",
    "type": "text",
  },
  "placeOfBirth": {
    "answer": "test",
    "category": "applicantDetails",
    "key": "placeOfBirth",
    "title": "Place of birth ",
    "type": "text",
  },
  "service": {
    "answer": "msc",
    "category": "documents",
    "key": "service",
    "title": "Service being applied for",
    "type": "list",
  },
  "surname": {
    "answer": "test",
    "category": "applicantDetails",
    "key": "surname",
    "title": "Surname",
    "type": "text",
  },
}
`;
